pkg/tfbridge/schema.go

```diff
@@ -461,13 +461,27 @@ const metaKey = "__meta"
 // MakeTerraformResult expands a Terraform state into an expanded Pulumi resource property map.  This respects
 // the property maps so that results end up with their correct Pulumi names when shipping back to the engine.
 func MakeTerraformResult(state *terraform.InstanceState,
-	tfs map[string]*schema.Schema, ps map[string]*SchemaInfo) resource.PropertyMap {
+	tfs map[string]*schema.Schema, ps map[string]*SchemaInfo) (resource.PropertyMap, error) {
 	var outs map[string]interface{}
 	if state != nil {
 		outs = make(map[string]interface{})
 		attrs := state.Attributes
+
+		reader := &schema.MapFieldReader{
+			Schema: tfs,
+			Map:    schema.BasicMapReader(attrs),
+		}
 		for _, key := range flatmap.Map(attrs).Keys() {
-			outs[key] = flatmap.Expand(attrs, key)
+			res, err := reader.ReadField([]string{key})
+			if err != nil {
+				return nil, err
+			}
+			if res.Value != nil {
+				outs[key] = res.Value
+			}
+		}
+		if _, ok := outs["id"]; !ok {
```

Comment:
Perhaps a comment on why this is needed?

Reply:
Sure.